src/doc/src/reference/build-scripts.md

```diff
@@ -250,6 +254,39 @@ identifier, the value should be a string.
 [conditional compilation]: ../../reference/conditional-compilation.md
 [option-cfg]: ../../rustc/command-line-arguments.md#option-cfg
 
+### `cargo::rustc-check-cfg=CHECK_CFG` {#rustc-check-cfg}
+
+The `rustc-check-cfg` instruction tells Cargo to pass the given expected
+configs to the [`--check-cfg` flag][option-check-cfg] to the compiler. This is
+used for compile-time detection of unexpected [conditional compilation] configs.
+
+You can use the instruction like this:
+
+```rust,no_run
+// build.rs
+println!("cargo::rustc-check-cfg=cfg(foo, values(\"bar\"))");
+```
+
+Be aware that the [`--check-cfg` flag][option-check-cfg] expected all possible
+configs, regardless of which is currently enabled to **always** be passed.
```

Comment:
```suggestion
Note that all possible
cfgs and values should be defined, regardless of what cfgs or values are currently active
```
There were some odd pieces of grammar (e.g. "expected" instead of "expects") but in general felt the structure was odd.

Reply:
It was indeed very odd. Fixed using your suggestion as baseline.

Note that I changed some terminology, "active" -> "enabled" and avoided the "cfg and values" as cfg generally means "key=value", and instead put the values warning in it's own sentence.

<details>

> Note that all possible cfgs should be defined, regardless of which cfgs are currently enabled. This includes all possible values of a given cfg name.

</details>